os/binfmt/binfmt_loadbinary.c

```diff
@@ -160,6 +164,24 @@ int load_binary(int binary_idx, FAR const char *filename, load_attr_t *load_attr
 			goto errout_with_bin;
 		}
 
+#ifdef CONFIG_DEBUG_BIN_SYMBOL
+		bin_addr_info_t *bin_info;
+		bin_info = (bin_addr_info_t *)kmm_malloc(sizeof(bin_addr_info_t));
+		if (bin_info != NULL) {
+			strncpy(bin_info->bin_name, bin->bin_name, BIN_NAME_MAX);
+			bin_info->text_addr = (uint32_t)bin->alloc[ALLOC_TEXT];
+			bin_info->rodata_addr = (uint32_t)bin->alloc[ALLOC_DATA];
+			bin_info->data_addr = (uint32_t)bin->datastart;
+			bin_info->bss_addr = (uint32_t)bin->bssstart;
+
+			binfo("[%s] text_addr : %x, rodata_addr : %x, data_addr : %x, bss_addr : %x\n", bin_info->bin_name, bin_info->text_addr, bin_info->rodata_addr, bin_info->data_addr, bin_info->bss_addr);
+
+			sq_addlast((sq_entry_t *)bin_info, &g_bin_addr_list);
+		} else {
+			berr("ERROR : Failed to allocate for bin symbol.\n");
+		}
+#endif
+
```

Comment:
There is still a chance of error in line 203 and 226.
How about moving this after line 235?
We can move the return statements also after line 235.